components/InstallButton.tsx

```diff
@@ -0,0 +1,94 @@
+'use client';
+
+import { ChromeIcon } from '@/public/assets/misc_icons/';
+import { ChevronDownIcon } from '@heroicons/react/20/solid';
+import { Dispatch, SetStateAction, ReactNode, useState } from 'react';
+import { useActiveLinkContext } from './ActiveLinkContext';
+
+interface BrowserSelectPopupTypes {
+  options: OptionType[];
+  setActiveLink: Dispatch<SetStateAction<any>>;
+  isBrowserPopupActive: boolean;
+  setIsBrowserPopupActive: Dispatch<SetStateAction<any>>;
+}
+
+interface OptionType {
+  id: number;
+  buttonContent: {
+    svg: ReactNode;
+    name: string;
+    linkRedirect: string;
+  };
+}
+
+const options: OptionType[] = [
+  {
+    id: 1,
+    buttonContent: {
+      svg: <ChromeIcon />,
+      name: 'Chrome',
+      linkRedirect:
+        'https://chromewebstore.google.com/detail/chatgpt-styler/mmfefeellkogplnodhfnggcpkomlaapj?hl=en&authuser=0&pli=1',
+    },
+  },
+];
+
+export default function InstallButton() {
+  const [isBrowserPopupActive, setIsBrowserPopupActive] =
+    useState<boolean>(false);
+  const { activeLink, setActiveLink } = useActiveLinkContext();
+
+  return (
+    <div className='flex justify-between text-white rounded-md w-60 h-11 px-2 py-1.5 items-center bg-teal-500 shadow-lg shadow-teal-800/50 relative'>
+      <button
+        onClick={() => window.open(activeLink.linkRedirect, '_blank')}
+        className='flex items-center gap-2'
+      >
+        {activeLink.svg}
+        {`Install for ${activeLink.name}!`}
+      </button>
+      <div className='ml-2 w-0.5 h-full border bg-white'></div>
+      <button onClick={() => setIsBrowserPopupActive(!isBrowserPopupActive)}>
+        <ChevronDownIcon className='w-8 h-8 font-thin hover:text-teal-800/50' />
+      </button>
+
+      <BrowserSelectPopup
+        options={options}
+        setActiveLink={setActiveLink}
+        isBrowserPopupActive={isBrowserPopupActive}
+        setIsBrowserPopupActive={setIsBrowserPopupActive}
+      />
+    </div>
+  );
+}
+
+const BrowserSelectPopup = ({
+  options,
+  setActiveLink,
+  isBrowserPopupActive,
+  setIsBrowserPopupActive,
+}: BrowserSelectPopupTypes) => {
+  const items = options.map((i) => (
```

Comment:
```suggestion
  const items = options.map((option: OptionType, index: number) => (
```

then adjust the type and remove `id`